pkg/scheduler/scheduler.go

```diff
@@ -310,12 +311,12 @@ func NewFromConfig(config *factory.Config) *Scheduler {
 }
 
 // Run begins watching and scheduling. It waits for cache to be synced, then starts a goroutine and returns immediately.
-func (sched *Scheduler) Run() {
+func (sched *Scheduler) Run(ctx context.Context) {
 	if !sched.WaitForCacheSync() {
 		return
 	}
 
-	go wait.Until(sched.scheduleOne, 0, sched.StopEverything)
+	go wait.UntilWithContext(ctx, sched.scheduleOne, 0)
```

Comment:
Can we get ride of `sched.StopEverything` altogether?

Reply:
I think we could get rid of that in another PR.